fish-rust/src/builtins/shared.rs

```diff
@@ -86,9 +86,9 @@ impl output_stream_t {
         unsafe { (*self.0).pin() }
     }
 
-    /// Append a &wtr or WString.
-    pub fn append<Str: AsRef<wstr>>(&mut self, s: Str) -> bool {
-        self.ffi().append1(c_str!(s))
+    /// Append a &wstr or WString.
+    pub fn append<Str: AsRef<wstr> + ToCppWString>(&mut self, s: Str) -> bool {
+        self.ffi().append(&s.into_cpp())
```

Comment:
```suggestion
        self.ffi().append(&s.as_ref().into_cpp())
```
Then you don't need the bound on `ToCppWString`, which saves some implementations when using e.g. `Cow<`a, wstr>`

Reply:
Thanks, implemented!